applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -173,6 +174,31 @@ Vector SmallStrainUPwDiffOrderElement::GetPressures(const size_t n_nodes) const
     return pressure;
 }
 
+void set_arithmetic_average_pressure(Geometry<Node>&                               rGeometry,
+                                     const Vector&                                 rPressure,
+                                     const std::vector<std::pair<size_t, size_t>>& rIndexPpairs,
+                                     size_t DestinationOffset = 0)
+{
+    for (size_t i = 0; const auto& [first_index, second_index] : rIndexPpairs) {
+        NodeUtilities::ThreadSafeNodeWrite(rGeometry[DestinationOffset + i], WATER_PRESSURE,
+                                           0.5 * (rPressure[first_index] + rPressure[second_index]));
+        ++i;
+    }
+}
+
+void set_arithmetic_average_pressure(Geometry<Node>& rGeometry,
```

Comment:
Same as previous comment
```suggestion
void SetArithmeticAveragePressure(Geometry<Node>& rGeometry,
```